Sage/carbon
# docs: replace Formik/Yup example and add explanatory notes

### Proposed behaviour

The docs on Validation should link to a working example of Carbon with Formik and Yup validation, and clearly explain any important points.



### Current behaviour

The linked sandbox crashes when updating almost any of the component values. No explanation is given of the CSB example despite some of the code being quite complex.



### Checklist




#### QA


### Additional context



### Testing instructions



Check the documentation added in the `validation.stories.mdx` file - suggest any changes necessary to make it more readable and understandable, or any key points missed.

Also carefully inspect/test the codesandbox to ensure the validation of each component behaves as desired and that there are no other bugs.
